src/_DataStructures_/Trees/BST/index.js

```diff
@@ -82,29 +82,123 @@ class BinarySearchTree {
     }
     return false;
   }
+
+  delete(root, value) {
+    if (root === null) {
+      return root;
+    }
+
+    if (value > root.value) {
+      // eslint-disable-next-line no-param-reassign
+      root.rightChild = this.delete(root.rightChild, value);
+    } else if (value < root.value) {
+      // eslint-disable-next-line no-param-reassign
+      root.leftChild = this.delete(root.leftChild, value);
+    } else {
+      // found the node
+      if (root.leftChild === null) {
+        // there is a right sub-tree
+        return root.rightChild;
+      }
+      if (root.rightChild === null) {
+        // there is a left sub-tree
+        return root.leftChild;
+      }
+      // the root contain 2 childs
+      const minRightNode = this.findMinNode(root.rightChild);
+      // eslint-disable-next-line no-param-reassign
+      root.value = minRightNode.value;
+      // eslint-disable-next-line no-param-reassign
+      root.rightChild = this.delete(root.rightChild, minRightNode.data);
```

Comment:
`minRightNode.value` not `minRightNode.data`.

Reply:
fixed 😖